### PR TITLE
fix: allow custom value for cpu and ram

### PR DESCRIPTION
**Pull request by minz1027**
- allow custom value for cpu and ram so that user with special needs can custom their build resources.
- there is a configurable upper bound (will be added here later: https://github.com/screwdriver-cd/buildcluster-queue-worker/blob/master/config/custom-environment-variables.yaml#L18)

- will figure out how to prevent abuse later

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -239,7 +239,12 @@ class K8sVMExecutor extends Executor {
             LOW: this.lowCpu,
             MICRO: this.microCpu
         };
-        const CPU = (cpuConfig in cpuValues) ? cpuValues[cpuConfig] : cpuValues.LOW;
+        let cpu = (cpuConfig in cpuValues) ? cpuValues[cpuConfig] : cpuValues.LOW;
+
+        // allow custom cpu value
+        if (Number.isInteger(cpuConfig)) {
+            cpu = cpuConfig;
```
Comment (jithin1987) [inline index.js]: lets have an upper threshold of 32.

File: index.js [hunk 2]
```diff
@@ -249,7 +254,12 @@ class K8sVMExecutor extends Executor {
         };
         const memConfig = annotations[RAM_RESOURCE];
         const MEMORY_GB = (memConfig in memValues) ? memValues[memConfig] : memValues.LOW;
-        const MEMORY = MEMORY_GB * 1024;
+        let memory = MEMORY_GB * 1024;
+
+        // allow custom memory value
+        if (Number.isInteger(memConfig)) {
+            memory = memConfig * 1024;
```
Comment (jithin1987) [inline index.js]: lets have an upper threshold of 65G

File: index.js (hunk 1 above)
Comment (minz1027) [reply index.js]: set the default max as turbo value. we will override that in our internal worker config

File: index.js [hunk 3]
```diff
@@ -239,7 +243,12 @@ class K8sVMExecutor extends Executor {
             LOW: this.lowCpu,
             MICRO: this.microCpu
         };
-        const CPU = (cpuConfig in cpuValues) ? cpuValues[cpuConfig] : cpuValues.LOW;
+        let cpu = (cpuConfig in cpuValues) ? cpuValues[cpuConfig] : cpuValues.LOW;
+
+        // allow custom cpu value
+        if (Number.isInteger(cpuConfig)) {
+            cpu = cpuConfig > this.maxCpu ? this.maxCpu : cpuConfig;
```
Comment (jithin1987) [inline index.js]: `cpu = Math.min(cpuConfig, this.maxCpu)` is concise. Same for memory as well.

Comment (minz1027) [reply index.js]: good suggestion! addressed

**Pull request merged**